### DISCUSSION TITLE
SwitchMode 15 doesn't send switch state when reconnecting to MQTT

### DISCUSSION THREAD

**Discussion by boozelclark** [arendst/Tasmota#12444] [Q&A] [answered]
Hi

I am running a Shelly 1 with Tasmota with SwitchMode 15 so I can use the Output relay and the switch independently. The problem I have is that if the MQTT broker and home assistant both restart then the Tasmota device does not send the switch state when it reconnects. It does send the Sensor state however. Is there a way to have Tasmota send the relay state when reconnecting? It does send it as soon as I restart the Tasmota device

**Comment by barbudor** [comment, marked as answer]
With `SwitchMode 15` you should get the switch status in the teleperiod SENSOR message, as well as any time the switch change state.
You could force Tasmota to resend a tele/SENSOR message on MQTT reconnect using a [rule](https://tasmota.github.io/docs/Rules/) such as
```
ON mqtt#connected DO teleperiod ENDON
```

**Reply by boozelclark** [reply]
Hi. I do get the switch state but not the relay state.

**Reply by barbudor** [reply]
Ok, if the problem is power and not switch:
```
ON mqtt#connected DO power ENDON
```

**Reply by boozelclark** [reply]
Thanks very much! I'll give it a try. Would PowerRetain then have a similar effect?

**Reply by barbudor** [reply]
PowerRetain will store the lastest Power state in the broker.
If the MQTT broker keeps retained values across restart, then it may cover most cases too and Tasmota won't need to send again the POWER state. However, if the Tasmota device power state changed while the MQTT broker is restarting, then the broker will not be informed of the new state.
PowerRetain is more targeted to "clients reconnect" like H.A. : Imagine that the MQTT broker do not restart but H.A. does, then PowerRetain will make the MQTT broker provide to H.A. the last valid value when H.A. re-subscribe to the topic.

The rule will address re-updating the MQTT broker "memory" (or its clients) of any state that may have changed while the broker was "away".

So both have an interest and address different failure cases.

**Reply by boozelclark** [reply]
Thanks for the assistance! I have raised an issue for it https://github.com/arendst/Tasmota/issues/12458 I'll do some more detailed testing on my side too when i have a chance. Thanks very much

**Reply by barbudor** [reply]
I remove my comment as after some investigation, the reason for that cmnd/.../POWER in the reconnection is specific for some brokers
The advised rule is the way to go
